### PR TITLE
Idiomatic checks in model config checking

### PR DESCRIPTION
**Pull request by tisnik**
## Type of change

- [x] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Description

Idiomatic checks in model config checking

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- N/A

### CODE REVIEW

File: app/models/config.py [hunk 1]
```diff
@@ -99,11 +99,11 @@ def __init__(self, data: Optional[dict] = None):
...
-            if data.get("redis", None) is None:
+            if "redis" not in data:
                 raise Exception("redis config is missing")
-            self.redis = RedisConfig(data.get("redis", None))
+            self.redis = RedisConfig(data["redis"])
         elif self.type == "in-memory":
-            if data.get("in-memory", None) is None:
+            if "in-memory" not in data:
                 raise Exception("in-memory config is missing")
             self.memory = MemoryConfig(data.get("memory", None))
```
Comment (tisnik) [inline app/models/config.py:108]: @bparees here I'm not sure if we really want to access `data["memory"]` or `data["in-memory"]`

Comment (bparees) [reply app/models/config.py:108]: `memory` is correct here.

here is an example of the config section:
```
  conversation_cache:
    type: in-memory
    memory:
      max_entries: 1000
```

you can change `in-memory` to `memory` and be consistent everywhere, i think @vishnuchalla persuaded me to use `in-memory` at some point but i'm starting to doubt that choice, i think `memory` is sufficiently clear relative to the other option (some sort of persistent db storage)

cc @syedriko

Comment (bparees) [reply app/models/config.py:108]: i'll merge this as is and leave it up to @syedriko if he wants to change `in-memory` to `memory` in places where we currently use `in-memory`

**Pull request merged**